### ISSUE TITLE
Crash on conversations with no messages using Layer 0.23.1

### ISSUE THREAD

**Issue by Stakenborg** [layerhq/Atlas-iOS#1259]
This message is printed to console just before crash:

`Zero is not a valid value for the pagination window.`

Happens during initialization of `ATLConversationDataSource`. Conversations with old messages in them seem to be fine.

**Comment by xs2bush** [comment]
Having the same problem. Its happening because of the following API change in 0.23.1:

> Setting the paginationWindow of an LYRQueryController object to zero will now raise an NSInvalidArgumentException.

When the ATLConversationViewController viewDidLoad is called, the LYRQueryController used to fetch the messages is set with the paginationWindow = numberOfMessagesToDisplay in the conversation. For new conversations, this number is zero which causes the crash. See ATLConversationViewController line 253 and trace it to ATLConversationDataSource line 77. 

I changed it to 

`_queryController.paginationWindow = numberOfMessagesToDisplay == 0 ? 1 : -numberOfMessagesToDisplay;`

and the crash goes away. I haven't tested it well to submit a pull request. Not even sure if this is a good solution.

**Comment by daniel-maness** [comment]
Have you updated to Atlas 1.0.28 yet? It includes [this commit](https://github.com/layerhq/Atlas-iOS/commit/d6d515595492798775c428e383c2d04b50ab77e1) that fixes that crash.

**Comment by Stakenborg** [comment]
I did notice that, and currently have my podfile point to the 1.0.28 release. The podspec is still outdated, however, so if anyone follows the recommended Cocoapods installation, they're going to run into this issue.

**Comment by xs2bush** [comment]
@Stakenborg did you get this error:

`[!] Unable to satisfy the following requirements:

- `Atlas (= 1.0.28)` required by `Podfile`

None of your spec sources contain a spec satisfying the dependency: `Atlas (= 1.0.28)`.

You have either:
* out-of-date source repos which you can update with `pod repo update`.
* mistyped the name or version.
* not added the source repo that hosts the Podspec to your Podfile.

Note: as of CocoaPods 1.0, `pod repo update` does not happen on `pod install` by default.`

I got this and doing `pod repo install` before `pod update` doesnt help. I tried deleting the cocoapods repo, clear cache and then `pod setup` but no luck. Let me know if you find a solution

**Comment by elitalon** [comment]
@xs2bush It's not a problem on your side. I guess that the updated version of `Atlas.podspec` hasn't been pushed to CocoaPods Trunk yet

**Comment by Stakenborg** [comment]
@xs2bush Until the podspec is updated on Cocoapods, you can pin directly to the latest release tag with this:
`pod 'Atlas', :git => 'https://github.com/layerhq/Atlas-iOS.git', :tag => 'v1.0.28'`

Or if you just want to pin to master for now, drop the tag part:
`pod 'Atlas', :git => 'https://github.com/layerhq/Atlas-iOS.git'`

Good enough I guess :: shrug ::

**Comment by bimusiek** [comment]
I am having similar issue using
```
  - Atlas (1.0.27):
    - LayerKit (>= 0.22.0)
```

However in my case, `numberOfMessagesAvailable` is `nil`, which make this code
```obj-c
NSUInteger numberOfMessagesToDisplay = MIN(numberOfMessagesAvailable, ATLQueryControllerPaginationWindow);
```
return 0 which generates this exception
```
2016-09-26 10:53:13.330 Ahoy[40793:1787463] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Zero is not a valid value for the pagination window.'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000011241634b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x0000000111e7721e objc_exception_throw + 48
	2   CoreFoundation                      0x000000011247f265 +[NSException raise:format:] + 197
	3   LayerKit                            0x000000010e2444ab -[LYRQueryController setPaginationWindow:] + 61
	4   Atlas                               0x000000010c4a7883 -[ATLConversationDataSource initWithLayerClient:query:] + 659
	5   Atlas                               0x000000010c4a75bb +[ATLConversationDataSource dataSourceWithLayerClient:query:] + 123
	6   Atlas                               0x000000010c4b4e00 -[ATLConversationViewController fetchLayerMessages] + 960
	7   Atlas                               0x000000010c4b48cd -[ATLC2016-09-26 10:53:13.330824 Ahoy[40793:1819905] [] nw_resolver_cancel_on_queue 0x618000300480
onversationViewController setConversation:] + 349
	8   Ahoy                                0x000000010b9ca84c _TFC4Ahoy11LayerChatVC12setUpLayerVCfCSo15LYRConversationT_ + 1324
	9   Ahoy                                0x000000010b9c9a93 _TFC4Ahoy11LayerChatVC17onGotConversationfCSo15LYRConversationT_ + 147
	10  Ahoy                                0x000000010b9c8137 _TFFC4Ahoy11LayerChatVC11viewDidLoadFT_T_U_FV10Foundation12NotificationT_ + 759
	11  Ahoy                                0x000000010b8d0844 _TTRXFo_iV10Foundation12Notification__XFo_iS0__iT__ + 20
	12  Ahoy                                0x000000010b9c82e1 _TPA__TTRXFo_iV10Foundation12Notification__XFo_iS0__iT__ + 81
	13  RxSwift                             0x000000010f6a1966 _TFFE7RxSwiftPS_14ObservableType9subscribeFT6onNextGSqFwx1ET__7onErrorGSqFPs5Error_T__11onCompletedGSqFT_T__10onDisposedGSqFT_T___PS_10Disposable_U_FGOS_5EventQQPS0_1E_T_ + 806
	14  RxSwift                             0x000000010f6a1e0f _TPA__TFFE7RxSwiftPS_14ObservableType9subscribeFT6onNextGSqFwx1ET__7onErrorGSqFPs5Error_T__11onCompletedGSqFT_T__10onDisposedGSqFT_T___PS_10Disposable_U_FGOS_5EventQQPS0_1E_T_ + 239
	15  RxSwift                             0x000000010f607afa _TFC7RxSwift17AnonymousObserver6onCorefGOS_5Eventx_T_ + 378
	16  RxSwift                             0x000000010f6a960e _TFC7RxSwift12ObserverBase2onfGOS_5Eventx_T_ + 958
	17  RxSwift                             0x000000010f6a9b47 _TTWurGC7RxSwift12ObserverBasex_S_12ObserverTypeS_FS1_2onfGOS_5Eventwx1E_T_ + 55
	18  RxSwift                             0x000000010f6a74d5 _TFFC7RxSwift32ObserveOnSerialDispatchQueueSinkcFT9schedulerCS_28SerialDispatchQueueScheduler8observerx_GS0_x_U_FTGS0_Q__GOS_5EventQQ_1E__PS_10Disposable_ + 517
	19  RxSwift                             0x000000010f6a7704 _TPA__TFFC7RxSwift32ObserveOnSerialDispatchQueueSinkcFT9schedulerCS_28SerialDispatchQueueScheduler8observerx_GS0_x_U_FTGS0_Q__GOS_5EventQQ_1E__PS_10Disposable_ + 20
	20  RxSwift                             0x000000010f6a779d _TTRGRx7RxSwift12ObserverTyperXFo_oGCS_32ObserveOnSerialDispatchQueueSinkx_iGOS_5Eventwx1E__iPS_10Disposable__XFo_iTGS1_x_GS2_wxS3____iPS4___ + 141
	21  RxSwift                             0x000000010f6a785d _TPA__TTRGRx7RxSwift12ObserverTyperXFo_oGCS_32ObserveOnSerialDispatchQueueSinkx_iGOS_5Eventwx1E__iPS_10Disposable__XFo_iTGS1_x_GS2_wxS3____iPS4___ + 109
	22  RxSwift                             0x000000010f6a7f1c _TTRGRx7RxSwift12ObserverTyperXFo_iTGCS_32ObserveOnSerialDispatchQueueSinkx_GOS_5Eventwx1E___iPS_10Disposable__XFo_oGS1_x_iGS2_wxS3___iPS4___ + 412
	23  RxSwift                             0x000000010f6a7fd9 _TPA__TTRGRx7RxSwift12ObserverTyperXFo_iTGCS_32ObserveOnSerialDispatchQueueSinkx_GOS_5Eventwx1E___iPS_10Disposable__XFo_oGS1_x_iGS2_wxS3___iPS4___ + 89
	24  RxSwift                             0x000000010f6a779d _TTRGRx7RxSwift12ObserverTyperXFo_oGCS_32ObserveOnSerialDispatchQueueSinkx_iGOS_5Eventwx1E__iPS_10Disposable__XFo_iTGS1_x_GS2_wxS3____iPS4___ + 141
	25  RxSwift                             0x000000010f6a809d _TPA__TTRGRx7RxSwift12ObserverTyperXFo_oGCS_32ObserveOnSerialDispatchQueueSinkx_iGOS_5Eventwx1E__iPS_10Disposable__XFo_iTGS1_x_GS2_wxS3____iPS4___.8 + 109
	26  RxSwift                             0x000000010f686e68 _TFFC7RxSwift13MainScheduler16scheduleInternalurFTx6actionFxPS_10Disposable__PS1__U_FT_T_ + 184
	27  RxSwift                             0x000000010f61abc7 _TTRGrXFo___XFdCb___ + 39
	28  libdispatch.dylib                   0x000000011357d980 _dispatch_call_block_and_release + 12
	29  libdispatch.dylib                   0x00000001135a70cd _dispatch_client_callout + 8
	30  libdispatch.dylib                   0x00000001135878d6 _dispatch_main_queue_callback_4CF + 406
	31  CoreFoundation                      0x00000001123da4f9 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	32  CoreFoundation                      0x000000011239ff8d __CFRunLoopRun + 2205
	33  CoreFoundation                      0x000000011239f494 CFRunLoopRunSpecific + 420
	34  GraphicsServices                    0x0000000114198a6f GSEventRunModal + 161
	35  UIKit                               0x000000010ff1cf34 UIApplicationMain + 159
	36  Ahoy                                0x000000010bad9cff main + 111
	37  libdyld.dylib                       0x00000001135f368d start + 1
)
```

<img width="1920" alt="screenshot 2016-09-26 10 49 33" src="https://cloud.githubusercontent.com/assets/1025588/18828312/85d92fa6-83d7-11e6-922f-bfff9220ebc3.png">

**Comment by blakewatters** [comment]
This is fixed in >= v1.0.28 which is available on CocoaPods

**Comment by LukeDurrant** [comment]
FYI I had this crash on the latest version of master
